taraxa/state/dpos/precompiled/dpos_contract.go

```diff
@@ -792,17 +799,22 @@ func (self *Contract) undelegate(ctx vm.CallFrame, block types.BlockNum, args so
 		return ErrInsufficientDelegation
 	}
 
-	if delegation.Stake.Cmp(args.Amount) != 0 && self.cfg.MinimumDeposit.Cmp(bigutil.Sub(delegation.Stake, args.Amount)) == 1 {
+	if delegation.Stake.Cmp(args.Amount) != 0 && self.dpos_config.MinimumDeposit.Cmp(bigutil.Sub(delegation.Stake, args.Amount)) == 1 {
 		return ErrInsufficientDelegation
 	}
 
-	prev_vote_count := voteCount(validator.TotalStake, self.cfg.EligibilityBalanceThreshold, self.cfg.VoteEligibilityBalanceStep)
+	prev_vote_count := voteCount(validator.TotalStake, self.dpos_config.EligibilityBalanceThreshold, self.dpos_config.VoteEligibilityBalanceStep)
 
 	state, state_k := self.state_get(args.Validator[:], BlockToBytes(block))
 	if state == nil {
 		old_state := self.state_get_and_decrement(args.Validator[:], BlockToBytes(validator.LastUpdated))
 		state = new(State)
-		state.RewardsPer1Stake = bigutil.Add(old_state.RewardsPer1Stake, self.calculateRewardPer1Stake(validator_rewards.RewardsPool, validator.TotalStake))
+		if validator.TotalStake.Cmp(big.NewInt(0)) > 0 {
+			state.RewardsPer1Stake = bigutil.Add(old_state.RewardsPer1Stake, self.calculateRewardPer1Stake(validator_rewards.RewardsPool, validator.TotalStake))
+		} else {
+			state.RewardsPer1Stake = old_state.RewardsPer1Stake
```

Comment:
@MatusKysel pls check/confirm if these changes when I always check totalstake due to possible division by zero are ok. Is it ok work with the current "states architecture" ?